providers/curation/process.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get } = require('lodash')
+
+async function work() {
+  try {
+    let message = await queue.dequeue()
+    if (!get(message, 'data.pull_request') || !get(message, 'data.action')) return
+    const pr = message.data.pull_request
+    const action = message.data.action
+    switch (action) {
+      case 'opened':
+      case 'synchronize': {
+        const curations = await curationService.getContributedCurations(pr.number, pr.head.sha)
+        await curationService.validateContributions(pr.number, pr.head.sha, curations)
+        await curationService.updateContribution(pr, curations)
+        break
+      }
+      case 'closed': {
+        await curationService.updateContribution(pr)
+        break
+      }
+    }
+    logger.info(`Handled GitHub event "${action}" for PR#${pr.number}`)
+    await queue.delete(message)
+  } catch (error) {
+    logger.error(error)
+  } finally {
+    setTimeout(work, 30000)
+  }
+}
+
+let queue
+let curationService
+let logger
+
+function setup(_queue, _curationService, _logger) {
```

Comment:
Should we have any protection against someone calling this method twice?

Reply:
As it stands, call this method as many times as you want distinct workers to be pulling from the queue.

I feel like that behavior is appropriate, what do you think?